plugin/src/test/java/org/opensearch/ml/rest/RestSearchIndexToolIT.java

```diff
@@ -109,8 +110,8 @@ public void testSearchIndexToolInFlowAgent_withEmptyIndexField_thenThrowExceptio
         String exceptionMessage = exception.getMessage();
         int numNodes = NumberUtils.createInteger(System.getProperty("cluster.number_of_nodes", "1"));
         if (numNodes == 1 || !exceptionMessage.contains("RemoteTransportException")) {
-            // in multi-node cluster, the exception maybe of type RemoteTransportException
-            // we cannot get the concrete exception message from RemoteTransportException
+            // In multi-node cluster, the exception message may contain RemoteTransportException.
+            // Since exception is always of type exception, we cannot the detailed exception message.
```

Comment:
I think this is related with https://github.com/opensearch-project/ml-commons/issues/2443

@jngz-es, can you help fix the bug ? Then we can ask @yuye-aws to verify if this error is gone on multi-node cluster

Reply:
Thanks @jngz-es for creating this issue! We can wait merging this PR before https://github.com/opensearch-project/ml-commons/issues/2443